grpc/src/main/java/com/linecorp/armeria/server/grpc/HttpJsonTranscodingService.java

```diff
@@ -651,7 +651,7 @@ private static JsonNode getBodyContent(AggregatedHttpRequest request) {
         try {
             return mapper.readTree(request.contentUtf8());
         } catch (JsonProcessingException e) {
-            return null;
+            throw new IllegalArgumentException("error ocurred while parsing json request", e);
```

Comment:
I believe that makes sense, since it's part of the JsonTranscodingService, if there's any body, it should be a json type. I changed it so it throws an exception as well.

Reply:
How about raising an exception when the body is not null or empty?
I've also created a relevant issue. 😉 
#5311

Reply:
Empty node is invalid JSON, so I think it makes sense to throw an error too (currently it produces MissingNode)
I'm thinking the calling code, for the case where the body is `*`, it should be an error if the resulting node is not an ObjectNode regardless.
For the other case, I think we can allow all valid json, though NULL would likely result in a later error during proto conversion but I think that's better since then the json is at least valid.

Reply:
Sorry for making you get confused. 
What I meant was returning null when there's no content type and the body was empty here:
https://github.com/line/armeria/pull/5307/files#diff-a9d03eb106960c77af7b39790659dfa479f644ef0745c2e0075e56a5de8badacR648

Reply:
Ah yes, can do that, I think it makes sense.

I'm also thinking of throwing an exception here since it also defaults to an empty request if the resulting node is not ObjectNode:
https://github.com/line/armeria/blob/22163ca6dd7fa8d77e6f5104c486eda9b6018f79/grpc/src/main/java/com/linecorp/armeria/server/grpc/HttpJsonTranscodingService.java#L614C6-L614C6

Reply:
@minwoox can you take a look again?

Reply:
Shouldn't it be okay if it's null?
```java
if (body instanceof ObjectNode) {
    root = (ObjectNode) body;
} else if (body == null) {
    root = mapper.createObjectNode();
} else {
    throw new IllegalArgumentException("unexpected JSON: " + body +
                                       ", (expected: ObjectNode or null)");
}
```
Could you also add tests for cases that
- the request body is empty and content type is JSON
- the request body is empty and content type is null